src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -146,9 +154,19 @@ export const thorchainApi: SwapperApi = {
         const publicClient = viemClientByChainId[chainId as EvmChainId]
         assert(publicClient !== undefined, `no public client found for chainId '${chainId}'`)
 
+        const expectedAmountOut =
+          BigInt(longtailData?.longtailToLExpectedAmountOut ?? 0) || BigInt(0)
+        // Ensure we have this to prevent sandwich attacks on the first step of a LongtailToL1 trade.
+        assert(expectedAmountOut !== undefined, 'expectedAmountOut is undefined')
+
+        const amountOutMin = BigInt(
+          bnOrZero(expectedAmountOut.toString())
```

Comment:
A bit weird mixing BigInt primitive casted as string and `bignumber.js` but looks like they don't support BigInt so 🤷🏽

Reply:
Yeah I'm not a fan of this, but without built-in support I'm not sure how else to do it.